app/services/csv_exporter.rb

```diff
@@ -101,4 +130,33 @@ def dequote_and_truncate(text)
   def humanize_boolean(boolean)
     boolean ? 'Yes' : nil
   end
+
+  def num_days_late(kase)
+    if kase.date_draft_compliant.present? && kase.internal_deadline.present?
+      days = (kase.date_draft_compliant - kase.internal_deadline).to_i
+      days > 0 ? days : nil
+    end
+  end
+
+  # Caseworker officer is blank for non-trigger and
+  # always a disclosure specialist for trigger cases.
+  # Note Case#assigned_disclosure_specialist throws an exception if no
+  # 'approving' assignees are found
+  def casework_officer(kase)
```

Comment:
Both this method and the one below seem to know too much about the implementations of the methods they are calling. I think the functionality needs encapsulating - I don't think this code should know that trigger workflows result in assigned_disclosure_specialist not being valid?

Reply:
This was a requirement from the officer about how only to display a full_name if it's a trigger case. There's no suggestion that non-trigger cases should not have people assigned/names for display except for this particular report.

Reply:
This was a requirement from the officer about how only to display a full_name if it's a trigger case. There's no suggestion that non-trigger cases should not have people assigned/names for display except for this particular report.

Reply:
I think this points to issues with the kase#assigned_disclosure_specialist method. If there shouldn't be one (which is true for non-trigger cases) it seems find for this method to return nil rather than throw something nasty. That way the knowledge stays encapsulated within Case::Base where it belongs